src/DIRAC/WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -714,7 +715,12 @@ def _checkSubmittedJobs(self):
 
             # The payload failed (if result["Value"] is not 0)
             elif result["Value"]:
-                self.jobReport.setJobStatus(status=JobStatus.FAILED, minorStatus="Payload failed")
+                # In order to avoid overriding perfectly valid states, the status is updated iff the job was running
```

Comment:
Can you update the PR description with information about what prompted this change as I'm struggling to guess.

Reply:
Done.